.github/report_nightly_build_failure.py

```diff
@@ -0,0 +1,30 @@
+"""
+Called by GitHub Action when the nightly build fails.
+
+This reports an error to the #nightly-build-failures Slack channel.
+"""
+
+import os
+
+import requests
+
+if "SLACK_WEBHOOK_URL" in os.environ:
```

Comment:
This is an internal Slack channel. Not very visible... will probably re-asses at some point.

Reply:
I'm not against this. Integrations with common third party stuff is weird for sure. I've got some work related projects which have things like this for them, but also intend to release versions for other people. I've looked at several open source projects (both on purpose for this sort of thing and for other stuff where I just happened to notice this kind of integration) and it's not uncommon or unreasonable to have stuff like this as settings.

It definitely feels kind of weird to me to specifically support some things and not others even if they fill the same niche, but from what I've seen, it's not uncommon to support the ones you use.

Reply:
For sure, that's why this is good enough for now. I'm just considering that if failures were more visible, it would attract (new) contributors to help fix and prepare for whatever upstream changes we need to adapt to.